CHANGELOG.md

```diff
@@ -3,6 +3,43 @@ CHANGELOG
 
 ## HEAD (Unreleased)
 
+## IMPORTANT - COMPAT
+
+- Mitigate issue causing [crashes](https://github.com/pulumi/pulumi/issues/3260) and
+  [hangs](https://github.com/pulumi/pulumi/issues/3309) across many OSs (primarily CentOS and macOS)
+  and several versions of nodejs (primarily 12.12 and up).
+
+  The issue occurs when making a 'data source' call in Pulumi in a synchronous fashion.  i.e. code
+  like:
+
+  ```ts
+  // A call to some provider's `getXXX` data source function.
+  const lb = aws.lb.getLoadBalancer(...);
+  ```
+
+  The issue is mitigated such that it should occur much less for users in practice.  If your
+  data-source call does not pass in a `parent` or `provider` the issue should not appear anymore,
+  and you should not have to make any changes to your code.
+
+  If your data-source does pass in either of these values, this issue should only occur rarely.  If
+  it no longer occurs for you, you should not have to make any changes to your code.  However, if
+  you are still running into crashes or hangs on OSX, it is recommended to take the following
+  approach to fixing the issue:
+
+  ```ts
+  // Update the code where you create your provider from the following:
+  const provider = new aws.Provider(...);
+
+  // to:
+  const provider = await ProviderRef.get(new aws.Provider(...));
+  ```
+
+  This will now be a `ProviderRef` instead of a `Provider`.  However, it should be accepted with all
+  the latest Pulumi libraries anywhere a `Provider` was previously accepted.
+
+  In a future version, Pulumi libraries *may* be updated to no longer accept a `Provider` to help
+  ensure this issue doesn't occur at all.
+
```

Comment:
Let's simplify this significantly.   I think it can just say:

```
- Fix hangs and crashes related to use of `getResource` methods, including frequent hangs on Node 12.  Some less common existing styles of using `getResource` calls are also deprecated as part of this change, and users should see [link] for details on adjusting their code if needed.
```

We can move details into an issue or PR description or docs page (perhaps in Troubleshooting related to the warning text)?

Reply:
Docs here: https://github.com/pulumi/docs/pull/1852